WireBST.java

```diff
@@ -97,20 +97,94 @@ private void insertToRight(WireBSTNode current , WireBSTNode newNode ){
         newNode.setLeft(currentNode); // make the predecessor of the new node to be her parent, because it is the predecessor (only in insert to left!).
     }
 
-    // remove of the node WireBST
-    public void removeWireBST(WireBSTNode node){
 
+    /**
+     * remove of the node WireBST
+     * @param node is the node to remove.
+     */
+    public void removeWireBSTNode(WireBSTNode node){
+        if ( node == null) return;
+        if ( ! node.isRealLeft() && ! node.isRealRight() ){
+            removeDeepLeaf(node);
+        }
+        if ((node.getRight() != null) && (node.getleft() == null)){    //** in case of removing node who has only right son
```

Comment:
need to be: if ( node.isRealRight() && (! node.isRealLeft() ) )

no null in the tree except min & max :)